lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -1142,13 +1142,12 @@ class CleanupDxBreak : public FunctionPass {
     // For each wave operation, collect the blocks sensitive to it
     SmallPtrSet<BasicBlock *, 16> SensitiveBBs;
     for (Function &IF : M->functions()) {
-      if (&IF != &F && IF.getNumUses() && IF.isDeclaration() &&
-          (hlsl::GetHLOpcodeGroupByName(&IF) == HLOpcodeGroup::HLIntrinsic ||
-           hlsl::GetHLOpcodeGroup(&IF) == HLOpcodeGroup::HLExtIntrinsic)) {
+      HLOpcodeGroup opgroup = hlsl::GetHLOpcodeGroup(&IF);
+      if (&IF != &F && IF.getNumUses() && IF.isDeclaration() && IsHLWaveSensitive(&IF) &&
```

Comment:
Now the only problem I have is that getting the group is somewhat expensive, while some other tests are quite cheap that could early out, such as these: `&IF != &F && IF.getNumUses() && IF.isDeclaration()`.
Also '!IF.use_empty()` is cheaper than `IF.getNumUses()` (which has to iterate the uses to count them).
Not a blocker though.